xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -4082,10 +4082,16 @@ int CDVDPlayer::GetChapter()
   return m_State.chapter;
 }
 
-void CDVDPlayer::GetChapterName(std::string& strChapterName)
+void CDVDPlayer::GetChapterName(std::string& strChapterName, int chapterIdx)
 {
   CSingleLock lock(m_StateSection);
-  strChapterName = m_State.chapter_name;
+  if (chapterIdx == -1)
+    strChapterName = m_State.chapter_name;
+  else
+  {
+    if (m_pDemuxer)
+      m_pDemuxer->GetChapterName(strChapterName, chapterIdx);
```

Comment:
https://github.com/notspiff/kodi-cmake/commit/b964893b74c26219e0b35c4436528307fb3eb0b4 not tested.